source/features/close-out-of-view-modals.tsx

```diff
@@ -1,29 +1,58 @@
+import select from 'select-dom';
+import onetime from 'onetime';
+import delegate from 'delegate-it';
+
 import features from '.';
 
-const observer = new IntersectionObserver(([{intersectionRatio, target}]) => {
-	if (intersectionRatio === 0) {
-		observer.unobserve(target);
-		target.closest('details')!.open = false;
+const visible = new Set();
+const observer = new IntersectionObserver(entries => {
+	let lastModal: Element;
+	for (const {intersectionRatio, target: modal} of entries) {
+		if (intersectionRatio > 0) {
+			visible.add(modal);
+		} else {
+			visible.delete(modal);
+		}
```

Comment:
Too bad that `IntersectionObserver` doesn't keep a list of entries available at all times, so we have to track what's visible manually.